src/Windows/Avalonia.Win32/WindowImpl.AppWndProc.cs

```diff
@@ -336,6 +372,176 @@ protected virtual unsafe IntPtr AppWndProc(IntPtr hWnd, uint msg, IntPtr wParam,
+                                    for (int i = 0; i < historyCount - 1; i++)
+                                    {
+                                        var historyTouchInfo = historyTouchInfos[i];
+                                        var historyInfo = historyTouchInfo.pointerInfo;
+                                        var historyEventType = GetEventType(message, historyInfo);
+                                        var historyPoint = PointToClient(new PixelPoint(
+                                            historyInfo.ptPixelLocationX, historyInfo.ptPixelLocationY));
+                                        var historyModifiers = GetInputModifiers(historyInfo.dwKeyStates);
+                                        var historyTimestamp = historyInfo.dwTime == 0 ? timestamp : historyInfo.dwTime;
+                                        Input?.Invoke(new RawTouchEventArgs(_touchDevice, historyTimestamp, _owner,
+                                            historyEventType, historyPoint, historyModifiers, historyInfo.pointerId));
+                                    }
+                                }
+                            }
+                            else if (info.pointerType == PointerInputType.PT_PEN)
+                            {
+                                var pointerId = (uint)(ToInt32(wParam) & 0xFFFF);
+                                var historyCount = (int)info.historyCount;
+                                var historyPenInfos = new POINTER_PEN_INFO[historyCount];
+                                if (GetPointerPenInfoHistory(pointerId, ref historyCount, historyPenInfos))
+                                {
+                                    //last info is the same as the current so skip it
+                                    for (int i = 0; i < historyCount - 1; i++)
+                                    {
+                                        var historyPenInfo = historyPenInfos[i];
+                                        var historyInfo = historyPenInfo.pointerInfo;
+                                        var historyEventType = GetEventType(message, historyInfo);
+                                        var historyPoint = PointToClient(new PixelPoint(
+                                            historyInfo.ptPixelLocationX, historyInfo.ptPixelLocationY));
+                                        var historyModifiers = GetInputModifiers(historyInfo.dwKeyStates);
+                                        var historyTimestamp = historyInfo.dwTime == 0 ? timestamp : historyInfo.dwTime;
+
+                                        ApplyPenInfo(historyPenInfo);
+                                        Input?.Invoke(new RawPointerEventArgs(_penDevice, historyTimestamp, _owner,
+                                            historyEventType, historyPoint, historyModifiers));
+                                    }
+                                }
+                            }
```

Comment:
I guess GetPointerInfoHistory should be handled for mouse history.

Reply:
it always return only 1 item = useless for now. Maybe only 100k dpi gaming mouse can push items to history...
Ok I will handle it. no problem.

Reply:
It doesn't work for mouse, only for pen and touch

Reply:
Yeah, I also noticed it. And no, gaming mouse doesn't help here.

Reply:
Also
https://twitter.com/rickbrewPDN/status/1187496104478171136